From 5975ef49a3e072f750ef643e65b4ea322f240f1b Mon Sep 17 00:00:00 2001
From: Daniel Templeton <daniel@cloudera.com>
Date: Wed, 13 Jul 2016 13:26:35 -0700
Subject: [PATCH 1661/2848] Revert "YARN-4958. The file localization process
 should allow for wildcards to reduce the
 application footprint in the state store (Daniel
 Templeton via sjlee)"

This reverts commit 675412fc1729cfcc7cba336b2082e0cce4bcf003.

Change-Id: I5eb9ca6cb9cda574ab3428ff779872033e9b31b6
---
 .../yarn/server/nodemanager/ContainerExecutor.java |   15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/ContainerExecutor.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/ContainerExecutor.java
index 80712dd..77193df 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/ContainerExecutor.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/ContainerExecutor.java
@@ -50,7 +50,7 @@
 import org.apache.hadoop.util.StringUtils;
 
 public abstract class ContainerExecutor implements Configurable {
-  private static final String WILDCARD = "*";
+
   private static final Log LOG = LogFactory.getLog(ContainerExecutor.class);
   final public static FsPermission TASK_LAUNCH_SCRIPT_PERMISSION =
     FsPermission.createImmutable((short) 0700);
@@ -217,18 +217,7 @@ public void writeLaunchEnv(OutputStream out, Map<String, String> environment, Ma
     if (resources != null) {
       for (Map.Entry<Path,List<String>> entry : resources.entrySet()) {
         for (String linkName : entry.getValue()) {
-          if (new Path(linkName).getName().equals(WILDCARD)) {
-            // If this is a wildcarded path, link to everything in the
-            // directory from the working directory
-            File directory = new File(entry.getKey().toString());
-
-            for (File wildLink : directory.listFiles()) {
-              sb.symlink(new Path(wildLink.toString()),
-                  new Path(wildLink.getName()));
-            }
-          } else {
-            sb.symlink(entry.getKey(), new Path(linkName));
-          }
+          sb.symlink(entry.getKey(), new Path(linkName));
         }
       }
     }
-- 
1.7.9.5

